examples/benchmarks.py

```diff
@@ -409,13 +504,51 @@ def inference(inputs):
                         # To make sure that the model is traced + that the tensors are on the appropriate device
                         inference(sequence)
 
-                        runtimes = timeit.repeat(lambda: inference(sequence), repeat=average_over, number=3)
-                        average_time = sum(runtimes) / float(len(runtimes)) / 3.0
-                        dictionary[model_name]["results"][batch_size][slice_size] = average_time
+                        if not no_memory:
+                            # Line by line memory tracing (all code in the module `transformers`) works for all models/arbitrary code
+                            trace = start_memory_tracing("transformers")
+                            inference(sequence)
+                            summary = stop_memory_tracing(trace)
+
+                            if verbose:
+                                print(
+                                    "\nLines by line memory consumption:\n"
+                                    + "\n".join(
+                                        f"{frame.filename}:{frame.line_number}: mem {cpu_gpu_mem.string}: {frame.line_text}"
+                                        for frame, _, _, cpu_gpu_mem in summary.sequential
+                                    )
+                                )
+                                print(
+                                    "\nLines with top memory consumption:\n"
+                                    + "\n".join(
+                                        f"=> {frame.filename}:{frame.line_number}: mem {cpu_gpu_mem.string}: {frame.line_text}"
+                                        for frame, _, _, cpu_gpu_mem in summary.cumulative[:6]
+                                    )
+                                )
+                                print(
+                                    "\nLines with lowest memory consumption:\n"
+                                    + "\n".join(
+                                        f"=> {frame.filename}:{frame.line_number}: mem {cpu_gpu_mem.string}: {frame.line_text}"
+                                        for frame, _, _, cpu_gpu_mem in summary.cumulative[-6:]
+                                    )
+                                )
+                                print(f"\nTotal memory increase: {summary.total.string}")
+                            dictionary[model_name]["memory"][batch_size][slice_size] = summary.total.string
+                        else:
+                            dictionary[model_name]["memory"][batch_size][slice_size] = "N/A"
+
+                        if not no_speed:
+                            runtimes = timeit.repeat(lambda: inference(sequence), repeat=average_over, number=3)
```

Comment:
(nit) `number=3` feels like it should be exposed as a higher level kwarg like `nruns=3`

Reply:
Yeah that's the previous code. We can keep it as is for now.